File: java/1984.学生分数的最小差值.java
class Solution {
    // 1. O(nlogn) t:4ms(100%) O(logn) m:41.6M(5%)
    public int minimumDifference(int[] nums, int k) {
        Arrays.sort(nums);
        int ans = nums[k - 1] - nums[0];
        for (int i = 0; i <= nums.length - k; i++){
            int diff = nums[i + k - 1] - nums[i];
            if (diff < ans){
                ans = diff; 
            }
        }
        return ans;
    }
}

File: python/1984.学生分数的最小差值.py
class Solution:
    # 1. O(nlogn) t:40ms(83%) O(logn) m:15M(59%)
    def minimumDifference(self, nums: List[int], k: int) -> int:
        n = len(nums)
        if n == 1: return 0
        nums.sort()
        return min(nums[i + k - 1] - nums[i] for i in range(n - k + 1))